Tone/source/ExternalInput.js

```diff
@@ -0,0 +1,220 @@
+define(["Tone/core/Tone", "Tone/source/Source"], function(Tone){
+
+	"use strict";
+
+	/**
+	 *  @class  Tone.ExternalInput is a WebRTC Audio Input. Check 
+	 *          [Media Stream API Support](https://developer.mozilla.org/en-US/docs/Web/API/MediaStream_API)
+	 *          to see which browsers are supported. As of
+	 *          writing this, Chrome, Firefox, and Opera 
+	 *          support Media Stream. Chrome allows enumeration 
+	 *          of the sources, and access to device name over a 
+	 *          secure (HTTPS) connection. See [https://simpl.info](https://simpl.info/getusermedia/sources/index.html) 
+	 *          vs [http://simple.info](https://simpl.info/getusermedia/sources/index.html) 
+	 *          on a Chrome browser for the difference.
+	 *         
+	 *  @constructor
+	 *  @extends {Tone.Source}
+	 *  @param {number} [inputNum=0] If multiple inputs are present, select the input number. Chrome only.
+	 *  @example
+	 *  var motu = new Tone.ExternalInput(3);
+	 *  var soundflower = new Tone.ExternalInput().connect()
+	 *  
+	 *  Tone.ExternalInput.onload = function(){
+	 *  	motu.start();
+	 *  	selectSoundflower(Tone.ExternalInput.sourceList);
+	 *  }
+	 *  	
+	 *  function selectSoundflower(sources){
+	 *  	for(var i = 0; i < sources.length; i++){
+	 *  		if(sources[i].label === "soundflower"){
+	 *  			soundflowerIn.inputNum = i;
+	 *  			soundflowerIn.start();
+	 *  		}
+	 *  	}
+	 *  };
+	 */
+
+	Tone.ExternalInput = function(){
+
+		var options = this.optionsObject(arguments, ["inputNum", "onload"], Tone.ExternalInput.defaults);
```

Comment:
You never do anything with the "onload" parameter. Probably should be taken out.